### DISCUSSION TITLE
How to get the correct base type when generic?

### DISCUSSION THREAD

**Discussion by YBAZAN** [metalama/Metalama#49] [Q&A] [answered]
I use a TypeAspect on a generic class, namely `Entity<T>`

```C#
[TopAspect]
public class Entity<T>
{
...
}

public class ConcreteEntity : Entity<int>
{
    [ChildAspect]
    public string Name { get; set; }
}
```

From the `[ChildAspect]` inside the `ConcreteEntity` class, I would like to check if parent type has the `[TopAspect]` aspect. 
When I call the `builder.Target.BaseType.Aspects<TopAspect>()` method, I never get the aspect. 
I noticed that `builder.Target.BaseType` returns `Entity` as `INamedType`, not `Entity<T>`.

Is it a limitation? A bug? A bad use on my side?

**Comment by gfraiteur** [comment, marked as answer]
Use [INamedType.TypeDefinition](https://doc.metalama.net/api/metalama_framework_code_inamedtype_typedefinition#metalama_framework_code_inamedtype_typedefinition).

**Reply by YBAZAN** [reply]
You made my day!
